ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -521,4 +523,7 @@ default SnapshotContext getCurrentSnapshotContext(org.apache.hadoop.hive.ql.meta
   default void prepareAlterTableEnvironmentContext(AbstractAlterTableDesc alterTableDesc,
       EnvironmentContext environmentContext) {
   }
+  default List<String> getIcebergPartitions(Configuration conf, org.apache.hadoop.hive.ql.metadata.Table metaTable){
```

Comment:
should it be simply getPartitions and by default return UnsupportedOperationException if someone decides to use it with a different storage handler?

Reply:
Or what about giving back an optional to avoid exceptions?